scripts/github_meta.R

```diff
@@ -1,36 +1,55 @@
-# install.packages(c("yaml", "jsonlite", "devtools"))
-# devtools::install_github("cscheid/rgithub")
-# set environment variable GITHUB_TOKEN
+# install.packages(c("yaml", "jsonlite", "rlang", "gh"))
+# set environment variable GITHUB_PAT
 
 suppressMessages(library(yaml))
 suppressMessages(library(jsonlite))
-suppressMessages(library(github))
+suppressMessages(library(gh))
+
+`%||%` <- rlang::`%||%`
 
 message("* setting up context")
-ctx <- create.github.context(personal_token = Sys.getenv("GITHUB_TOKEN"))
 
 message("* loading yaml")
 yml <- yaml.load_file("_config.yml")
+widgets <- yml$widgets
+
+# use only packages on GitHub
+idx <- vapply(widgets, function(x) !is.null(x$ghuser), FUN.VALUE = logical(1))
+widgets <- widgets[idx]
+
+available_pkgs <- available.packages()[, "Package"]
 
-meta <- lapply(yml$widgets, function(wdgt) {
+meta <- lapply(widgets, function(wdgt) {
+  # if it's not hosted on GitHub, return 0
+  if (is.null(wdgt$ghuser))
+    return(list(stargazers_count = 0)) 
+  
   message("*** getting meta data for: ", wdgt$ghuser, " ", wdgt$ghrepo)
-  a <- try(get.repository(wdgt$ghuser, wdgt$ghrepo))
-  if(inherits(a, "try-error"))
+  
+  res <- try(gh("GET /repos/:owner/:repo", owner = wdgt$ghuser, repo = wdgt$ghrepo))
+
+  if(inherits(res, "try-error"))
     return(list(stargazers_count = 0))
-  a$content[c("stargazers_count", "open_issues_count", "forks_count", "watchers_count")]
+  
+  Sys.sleep(1)
+  
+  # check if the package is on CRAN
+  res$cran <- wdgt$name %in% available_pkgs
+
+  res[c("cran", "stargazers_count", "open_issues_count", "forks_count", "watchers_count")]
 })
 
-names(meta) <- sapply(yml$widgets, function(x)
+names(meta) <- sapply(widgets, function(x)
   paste(
     gsub("\\.", "_", x$ghuser),
     gsub("\\.", "_", x$ghrepo),
     sep = "_"))
 
-all_good <- sapply(meta, function(x) is.numeric(x$stargazers_count))
+all_good <- vapply(meta, function(x) is.numeric(x$stargazers_count), FUN.VALUE = logical(1))
 
-if(all(all_good) && length(yml$widgets) == length(meta)) {
+if(all(all_good) && length(widgets) == length(meta)) {
   message("* saving results")
-  cat(toJSON(meta, auto_unbox = TRUE), file = "github_meta.json")
+  cat(toJSON(meta, auto_unbox = TRUE, pretty = TRUE), file = "github_meta.json")
```

Comment:
I use `pretty = TRUE` because it's easier to see the diff, but no strong opinion.